api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -36,15 +54,113 @@
+            }
+            SearchRequest request = new Gson().fromJson(definition.getCriteria(), SearchRequest.class);
+            QueryResult result = bigQueryService.executeQuery(bigQueryService.filterBigQueryConfig(participantCounter.buildParticipantCounterQuery(request)));
+            Map<String, Integer> rm = bigQueryService.getResultMapper(result);
+            List<FieldValue> row = result.iterateAll().iterator().next();
+            long cohortCount = bigQueryService.getLong(row, rm.get("count"));
+
+            cohortReview = new CohortReview();
+            cohortReview.setCohortId(cohortId);
+            cohortReview.setCdrVersionId(cdrVersionId);
+            cohortReview.matchedParticipantCount(cohortCount);
+            cohortReview.setCreationTime(new Timestamp(System.currentTimeMillis()));
+            cohortReviewDao.save(cohortReview);
+        }
+
+        long count = participantCohortStatusDao.countByParticipantKey_CohortReviewId(cohortReview.getCohortReviewId());
+        CohortReviewInfo info = new CohortReviewInfo()
+                .cohortReviewId(cohortReview.getCohortReviewId())
+                .reviewCount(count)
+                .cohortCount(cohortReview.getMatchedParticipantCount());
+
+        return ResponseEntity.ok(info);
+    }
+
+    /**
+     * Create a cohort review per the specified cohortReviewId and size. If participant cohort status
+     * data exists for a review or no cohort review exists for cohortReviewId then throw a
+     * {@link BadRequestException}.
+     *
+     * @param cohortReviewId
+     * @param size
+     * @return
+     */
+    @Override
+    public ResponseEntity<CohortReviewInfo> createCohortReview(Long cohortReviewId, Integer size) {
+        if(participantCohortStatusDao.countByParticipantKey_CohortReviewId(cohortReviewId) > 0) {
+            throw new BadRequestException("Invalid Request: Cohort Review already created for cohortReviewId: " + cohortReviewId);
+        }
+        CohortReview cohortReview = cohortReviewDao.findOne(cohortReviewId);
+        if (cohortReview == null) {
+            throw new BadRequestException("Invalid Request: No Cohort Review matching cohortReviewId: " + cohortReviewId);
+        }
+
+        CohortDefinition definition = cohortDao.findCohortByCohortId(cohortReview.getCohortId());
+        SearchRequest request = new Gson().fromJson(definition.getCriteria(), SearchRequest.class);
+        QueryResult result = bigQueryService.executeQuery(bigQueryService.filterBigQueryConfig(participantCounter.buildParticipantIdQuery(request, size)));
+        Map<String, Integer> rm = bigQueryService.getResultMapper(result);
+
+        List<ParticipantCohortStatus> participantCohortStatuses = new ArrayList<>();
+        for (List<FieldValue> row : result.iterateAll()) {
+            participantCohortStatuses.add(
+                    new ParticipantCohortStatus()
```

Comment:
Do we need to set the status to NOT_REVIEWED?

Reply:
Done.